File: NetFramework/TouchToolkit/CS/TouchToolkitDemo/GcMagnifierPages/GcMagnifierCommon.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using C1.Win.TouchToolKit;

namespace ClickOnceDemo.C1MagnifyPages
{
    public partial class C1MagnifyCommon : DemoBase
    {
        public C1MagnifyCommon()
        {
            InitializeComponent();
            InitializeDemo();
        }

        private void InitializeDemo()
        {
            this.Title = "C1Magnify";
            this.Description = "This page shows C1Magnify.\n  -After long tap, a magnifier will popup above the finger, and shows the info around the finger.";

            moveLeftDownRadioButton.CheckedChanged += MoveMessageRadioButton_CheckedChanged;
            moveNoneRadioButton.CheckedChanged += MoveMessageRadioButton_CheckedChanged;
            moveRadioButton.CheckedChanged += MoveMessageRadioButton_CheckedChanged;

            closeLeftRadioButton.CheckedChanged += CloseMessageRadioButton_CheckedChanged;
            closeNoneRadioButton.CheckedChanged += CloseMessageRadioButton_CheckedChanged;
            closeRightRadioButton.CheckedChanged += CloseMessageRadioButton_CheckedChanged;

            circleRadioButton.CheckedChanged += ShapeRadioButton_CheckedChanged;
            rectangleRadioButton.CheckedChanged += ShapeRadioButton_CheckedChanged;
            roundedRectangleRadioButton.CheckedChanged += ShapeRadioButton_CheckedChanged;

            lightRadioButton.CheckedChanged += borderBackgroundModeRadioButton1_CheckedChanged;
            darkRadioButton.CheckedChanged += borderBackgroundModeRadioButton1_CheckedChanged;
            customRadioButton.CheckedChanged += borderBackgroundModeRadioButton1_CheckedChanged;

            ShowCursorCheckbox.DataBindings.Add("Checked",C1Magnify1,"ShowCursor");
            IsHorizontalRailEnabledCheckBox.DataBindings.Add("Checked", C1Magnify1, "IsHorizontalRailEnabled");
            IsVerticalRailEnalbedCheckBox.DataBindings.Add("Checked", C1Magnify1, "IsVerticalRailEnabled");

            if (C1Magnify1.BorderBackground != null)
            {
                pictureBox1.Image = C1Magnify1.BorderBackground;
            }
            else
            {
                C1Magnify1.BorderBackground = pictureBox1.Image;
            }

            borderWidthTrackBar.ValueChanged+=borderWidthTrackBar_ValueChanged;
            PopupDelayTrackBar.ValueChanged+=PopupDelayTrackBar_ValueChanged;
            sizeTrackBar.ValueChanged+=sizeTrackBar_ValueChanged;
            zoomfactorTrackBar.ValueChanged+=zoomfactorTrackBar_ValueChanged;
        }

        private void CloseMessageRadioButton_CheckedChanged(object sender, EventArgs e)
        {
            if (closeLeftRadioButton.Checked)
            {
                C1Magnify1.MessageWhenClose = MessageWhenClose.LeftClick;
            }
            else if (closeNoneRadioButton.Checked)
            {
                C1Magnify1.MessageWhenClose = MessageWhenClose.None;
            }
            else if (closeRightRadioButton.Checked)
            {
                C1Magnify1.MessageWhenClose = MessageWhenClose.RightClick;
            }
        }

        private void MoveMessageRadioButton_CheckedChanged(object sender, EventArgs e)
        {
            if (moveLeftDownRadioButton.Checked)
            {
                C1Magnify1.MessageWhenMove = MessageWhenMove.LeftDown;
            }
            else if (moveNoneRadioButton.Checked)
            {
                C1Magnify1.MessageWhenMove = MessageWhenMove.None;
            }
            else if (moveRadioButton.Checked)
            {
                C1Magnify1.MessageWhenMove = MessageWhenMove.Move;
            }
        }

        private void PopupDelayTrackBar_ValueChanged(object sender, EventArgs e)
        {
            popupDelayLabel.Text = PopupDelayTrackBar.Value.ToString() + "ms";
            C1Magnify1.PopupDelay = PopupDelayTrackBar.Value;
        }

        private void ShapeRadioButton_CheckedChanged(object sender, EventArgs e)
        {
            if (circleRadioButton.Checked)
            {
                C1Magnify1.Shape = MagnifierShape.Circle;
            }
            else if (rectangleRadioButton.Checked)
            {
                C1Magnify1.Shape = MagnifierShape.Rectangle;
            }
            else if (roundedRectangleRadioButton.Checked)
            {
                C1Magnify1.Shape = MagnifierShape.RoundedRectangle;
            }
        }

        private void sizeTrackBar_ValueChanged(object sender, EventArgs e)
        {
            C1Magnify1.Size = new Size(sizeTrackBar.Value*100, sizeTrackBar.Value*100);
            sizeLabel.Text = "(" + (sizeTrackBar.Value * 100).ToString() + "," + (sizeTrackBar.Value * 100).ToString() + ")";
        }

        private void zoomfactorTrackBar_ValueChanged(object sender, EventArgs e)
        {
            C1Magnify1.ZoomFactor = zoomfactorTrackBar.Value;
            zoomFactorLabel.Text = (zoomfactorTrackBar.Value * 100).ToString() + "%";
        }

        private void borderBackgroundModeRadioButton1_CheckedChanged(object sender, EventArgs e)
        {
            if (lightRadioButton.Checked)
            {
                C1Magnify1.BorderBackgroundMode = BorderBackground.Light;
                customBackgroundSetting.Visible = false;
            }
            else if (darkRadioButton.Checked)
            {
                C1Magnify1.BorderBackgroundMode = BorderBackground.Dark;
                customBackgroundSetting.Visible = false;
            }
            else if (customRadioButton.Checked)
            {
                C1Magnify1.BorderBackgroundMode = BorderBackground.Custom;
                customBackgroundSetting.Visible = true;
            }
        }

        private void SelectImageButton_Click(object sender, EventArgs e)
        {
            OpenFileDialog dialog = new OpenFileDialog();
            dialog.Filter = "Image Files(*.BMP;*.JPG;*.GIF;*.PNG)|*.BMP;*.JPG;*.GIF;*.PNG|All files (*.*)|*.*";
            if (DialogResult.OK == dialog.ShowDialog())
            {
                Bitmap bitmap = null;
                try
                {
                    bitmap = new Bitmap(dialog.FileName);
                }
                catch (ArgumentException)
                {
                    MessageBox.Show("This is not a suitable file.");
                    return;
                }
                if (C1Magnify1.BorderBackground != null)
                {
                    C1Magnify1.BorderBackground.Dispose();
                    C1Magnify1.BorderBackground = null;
                }
                C1Magnify1.BorderBackground = bitmap;
                pictureBox1.Image = bitmap;
            }
        }

        private void borderWidthTrackBar_ValueChanged(object sender, EventArgs e)
        {
            C1Magnify1.BorderWidth = borderWidthTrackBar.Value;
            borderWidthlabel.Text = C1Magnify1.BorderWidth.ToString();
        }
    }
}

File: NetFramework/TouchToolkit/VB/TouchToolkitDemo/GcMagnifierPages/GcMagnifierCommon.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports C1.Win.TouchToolKit

Namespace C1MagnifyPages
    Partial Public Class C1MagnifyCommon
        Inherits DemoBase
        Public Sub New()
            InitializeComponent()
            InitializeDemo()
        End Sub

        Private Sub InitializeDemo()
            Me.Title = "C1Magnify"
            Me.Description = "This page shows C1Magnify." & vbLf & "  -After long tap, a magnifier will popup above the finger, and shows the info around the finger."

            AddHandler moveLeftDownRadioButton.CheckedChanged, AddressOf MoveMessageRadioButton_CheckedChanged
            AddHandler moveNoneRadioButton.CheckedChanged, AddressOf MoveMessageRadioButton_CheckedChanged
            AddHandler moveRadioButton.CheckedChanged, AddressOf MoveMessageRadioButton_CheckedChanged

            AddHandler closeLeftRadioButton.CheckedChanged, AddressOf CloseMessageRadioButton_CheckedChanged
            AddHandler closeNoneRadioButton.CheckedChanged, AddressOf CloseMessageRadioButton_CheckedChanged
            AddHandler closeRightRadioButton.CheckedChanged, AddressOf CloseMessageRadioButton_CheckedChanged

            AddHandler circleRadioButton.CheckedChanged, AddressOf ShapeRadioButton_CheckedChanged
            AddHandler rectangleRadioButton.CheckedChanged, AddressOf ShapeRadioButton_CheckedChanged
            AddHandler roundedRectangleRadioButton.CheckedChanged, AddressOf ShapeRadioButton_CheckedChanged

            AddHandler lightRadioButton.CheckedChanged, AddressOf borderBackgroundModeRadioButton1_CheckedChanged
            AddHandler darkRadioButton.CheckedChanged, AddressOf borderBackgroundModeRadioButton1_CheckedChanged
            AddHandler customRadioButton.CheckedChanged, AddressOf borderBackgroundModeRadioButton1_CheckedChanged

            ShowCursorCheckbox.DataBindings.Add("Checked", gcMagnifier1, "ShowCursor")
            IsHorizontalRailEnabledCheckBox.DataBindings.Add("Checked", gcMagnifier1, "IsHorizontalRailEnabled")
            IsVerticalRailEnalbedCheckBox.DataBindings.Add("Checked", gcMagnifier1, "IsVerticalRailEnabled")

            If gcMagnifier1.BorderBackground IsNot Nothing Then
                pictureBox1.Image = gcMagnifier1.BorderBackground
            Else
                gcMagnifier1.BorderBackground = pictureBox1.Image
            End If

            AddHandler borderWidthTrackBar.ValueChanged, AddressOf borderWidthTrackBar_ValueChanged
            AddHandler PopupDelayTrackBar.ValueChanged, AddressOf PopupDelayTrackBar_ValueChanged
            AddHandler sizeTrackBar.ValueChanged, AddressOf sizeTrackBar_ValueChanged
            AddHandler zoomfactorTrackBar.ValueChanged, AddressOf zoomfactorTrackBar_ValueChanged
        End Sub

        Private Sub CloseMessageRadioButton_CheckedChanged(sender As Object, e As EventArgs)
            If closeLeftRadioButton.Checked Then
                gcMagnifier1.MessageWhenClose = MessageWhenClose.LeftClick
            ElseIf closeNoneRadioButton.Checked Then
                gcMagnifier1.MessageWhenClose = MessageWhenClose.None
            ElseIf closeRightRadioButton.Checked Then
                gcMagnifier1.MessageWhenClose = MessageWhenClose.RightClick
            End If
        End Sub

        Private Sub MoveMessageRadioButton_CheckedChanged(sender As Object, e As EventArgs)
            If moveLeftDownRadioButton.Checked Then
                gcMagnifier1.MessageWhenMove = MessageWhenMove.LeftDown
            ElseIf moveNoneRadioButton.Checked Then
                gcMagnifier1.MessageWhenMove = MessageWhenMove.None
            ElseIf moveRadioButton.Checked Then
                gcMagnifier1.MessageWhenMove = MessageWhenMove.Move
            End If
        End Sub

        Private Sub PopupDelayTrackBar_ValueChanged(sender As Object, e As EventArgs)
            popupDelayLabel.Text = PopupDelayTrackBar.Value.ToString() & "ms"
            gcMagnifier1.PopupDelay = PopupDelayTrackBar.Value
        End Sub

        Private Sub ShapeRadioButton_CheckedChanged(sender As Object, e As EventArgs)
            If circleRadioButton.Checked Then
                gcMagnifier1.Shape = MagnifierShape.Circle
            ElseIf rectangleRadioButton.Checked Then
                gcMagnifier1.Shape = MagnifierShape.Rectangle
            ElseIf roundedRectangleRadioButton.Checked Then
                gcMagnifier1.Shape = MagnifierShape.RoundedRectangle
            End If
        End Sub

        Private Sub sizeTrackBar_ValueChanged(sender As Object, e As EventArgs)
            gcMagnifier1.Size = New Size(sizeTrackBar.Value * 100, sizeTrackBar.Value * 100)
            sizeLabel.Text = "(" & (sizeTrackBar.Value * 100).ToString() & "," & (sizeTrackBar.Value * 100).ToString() & ")"
        End Sub

        Private Sub zoomfactorTrackBar_ValueChanged(sender As Object, e As EventArgs)
            gcMagnifier1.ZoomFactor = zoomfactorTrackBar.Value
            zoomFactorLabel.Text = (zoomfactorTrackBar.Value * 100).ToString() & "%"
        End Sub

        Private Sub borderBackgroundModeRadioButton1_CheckedChanged(sender As Object, e As EventArgs)
            If lightRadioButton.Checked Then
                gcMagnifier1.BorderBackgroundMode = BorderBackground.Light
                customBackgroundSetting.Visible = False
            ElseIf darkRadioButton.Checked Then
                gcMagnifier1.BorderBackgroundMode = BorderBackground.Dark
                customBackgroundSetting.Visible = False
            ElseIf customRadioButton.Checked Then
                gcMagnifier1.BorderBackgroundMode = BorderBackground.[Custom]
                customBackgroundSetting.Visible = True
            End If
        End Sub

        Private Sub SelectImageButton_Click(sender As Object, e As EventArgs) Handles SelectImageButton.Click
            Dim dialog As New OpenFileDialog()
            dialog.Filter = "Image Files(*.BMP;*.JPG;*.GIF;*.PNG)|*.BMP;*.JPG;*.GIF;*.PNG|All files (*.*)|*.*"
            If DialogResult.OK = dialog.ShowDialog() Then
                Dim bitmap As Bitmap = Nothing
                Try
                    bitmap = New Bitmap(dialog.FileName)
                Catch generatedExceptionName As ArgumentException
                    MessageBox.Show("This is not a suitable file.")
                    Return
                End Try
                If gcMagnifier1.BorderBackground IsNot Nothing Then
                    gcMagnifier1.BorderBackground.Dispose()
                    gcMagnifier1.BorderBackground = Nothing
                End If
                gcMagnifier1.BorderBackground = bitmap
                pictureBox1.Image = bitmap
            End If
        End Sub

        Private Sub borderWidthTrackBar_ValueChanged(sender As Object, e As EventArgs)
            gcMagnifier1.BorderWidth = borderWidthTrackBar.Value
            borderWidthlabel.Text = gcMagnifier1.BorderWidth.ToString()
        End Sub
    End Class
End Namespace